device/chip/chip.cpp

```
@@ -73,6 +73,10 @@ TLBManager* Chip::get_tlb_manager() {
         "Chip::get_tlb_manager is not available for this chip, it is only available for LocalChips.");
 }
 
+int Chip::get_num_host_channels() { return 0; }
+
+int Chip::get_host_channel_size(std::uint32_t channel) { return 0; }
```

I think this function should throw the exception. I am fine with returning 0 for num_host_channels, but for the size of the channel that doesn't exist, I would like to see the exception being thrown